File: Unspecified/sort_array_by_parity.py
from typing import List

class Solution:
    """
    Task:
    Given an integer array nums, move all the even integers at the beginning of the array followed by all the odd integers.

    Return any array that satisfies this condition.
    """
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        if len(nums) == 0:
            return nums
        writeIdx = 0
        for readIdx in range(len(nums)):
            if nums[readIdx] % 2 == 0:
                nums[readIdx], nums[writeIdx] = nums[writeIdx], nums[readIdx]
                writeIdx += 1
        return nums

File: Array&String/CPP/sort_array_by_parity.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        if (nums.size() <= 1) return nums;
        int i = 0;
        for (int j = 0; j < nums.size(); j++) {
            if (nums[j] % 2 == 0) {
                int temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                i++;
            }
        }
    return nums;
    }
};